lib/inputguess.rb

```diff
@@ -0,0 +1,44 @@
+class InputGuess
+  attr_reader :user_input, :valid_guess
+
+  def initialize
+    @valid_guess = false
+    @user_input = ""
+  end
+
+  def input_guess
+    until @valid_guess
+      print "Make a guess using available colors (R, O, Y, G, B, P): "
+      @user_input = clean_guess(gets.chomp)
+      if validate_length(@user_input)
+        validate_colors(@user_input)
+      else
+        puts "Please enter four colors using only their first letters."
+      end
+    end
+    @user_input
+  end
+
+  def clean_guess(guess)
+    begin
+      guess = guess.upcase.gsub(/[\W]/, "")
+    rescue NoMethodError, TypeError
+      puts "Please enter only letters."
+    end
+    guess
+  end
+
+  def validate_length(guess)
+    return true unless guess.length != 4
```

Comment:
`guess.length == 4`, works, no?

Reply:
Hmm, I think I need it to be true (length is only valid) if the input length is exactly 4, so anything that isn't 4 characters long should be false. Do you mean `return true if guess.length == 4`?

Reply:
`guess.length == 4` for the entire function. That evaluates to true or false. If it's more than 4 or less than four, it's false. If it's four, it's true and good.

Reply:
Yes! Done.